### PR TITLE
TAAR ensemble fix

### PR DESCRIPTION
**Pull request by crankycoder**
This fixes the TAAR ensemble job after the GCP migration.

The two major changes are:

* no more logging in recommenders as they are not serializable by python pickle until Python 3.8.  This caused crashes in PySpark workers
* added explicit AWS keys into a context object that is passed from the PySpark master to workers as environment variables are not properly configured to use the `airflow_taar_rw_s3` connection.

Both these changes are embedded in taar 0.4.12 which is imported by the DAG and are not visible in this PR.

I've added support for local SSDs to keep runtime down to ~30 minutes given 35 x n1-standard-8 nodes with 2 SSDs per node.

### CODE REVIEW

**Comment by crankycoder** [comment]
Output from a job run on a local Airflow instance can be found at:  s3://telemetry-parquet/taar/ensemble/ensemble_weight_daily20200322.json

```
{
  "ensemble_weights": {
    "collaborative": 2.4237995141351956,
    "similarity": 1.0295617216484558,
    "locale": 1.726434159924938
  }
}
```

File: dags/operators/moz_dataproc_operator.py [hunk 1]
```diff
@@ -91,6 +91,9 @@ class DataprocClusterCreateOperator(DataprocOperationBaseOperator):
     :type master_disk_type: str
     :param master_disk_size: Disk size for the master node
     :type master_disk_size: int
+    :param master_num_local_ssds : Number of local SSDs to mount
```
Comment (haroldwoo) [inline dags/operators/moz_dataproc_operator.py:94]: Please add a more descriptive comment on why we would ever want to set these. And what a defualt of 0 means

File: dags/utils/dataproc.py [hunk 2]
```diff
@@ -279,13 +288,19 @@ def moz_dataproc_pyspark_runner(parent_dag_name=None,
     :type master_disk_type: str
     :param master_disk_size:              Disk size for the master node
     :type master_disk_size: int
+    :param master_num_local_ssds : Number of local SSDs to mount
```
Comment (haroldwoo) [inline dags/utils/dataproc.py:291]: Thanks for adding this to all 3 runner methods. Can we add more description as to why we'd ever want to set these, and what a default of 0 means

File: dags/taar_weekly.py [hunk 3]
```diff
@@ -6,55 +6,77 @@
 from datetime import datetime, timedelta
 from airflow.operators.moz_databricks import MozDatabricksSubmitRunOperator
 from utils.mozetl import mozetl_envvar
+from airflow.operators.subdag_operator import SubDagOperator
```
Comment (haroldwoo) [inline dags/taar_weekly.py:9]: nit: imports arent grouped together or in any alphabetical ordering

File: dags/taar_weekly.py [hunk 4]
```diff
@@ -6,55 +6,77 @@
...
+from utils.dataproc import moz_dataproc_pyspark_runner
+from airflow.contrib.hooks.aws_hook import AwsHook
+
+taar_aws_conn_id = "airflow_taar_rw_s3"
+taar_aws_access_key, taar_aws_secret_key, session = AwsHook(
```
Comment (haroldwoo) [inline dags/taar_weekly.py:14]: nit: this fits on one line?

File: dags/taar_weekly.py [hunk 5]
```diff
@@ -6,55 +6,77 @@
...
-    external_dag_id="main_summary",
-    external_task_id="clients_daily_export",
-    execution_delta=timedelta(
-        days=-7, hours=-1
-    ),  # main_summary waits one hour, execution date is beginning of the week
-    dag=taar_weekly,
-)
-
-
-taar_ensemble = MozDatabricksSubmitRunOperator(
```
Comment (haroldwoo) [inline dags/taar_weekly.py:37]: rip db

File: dags/taar_weekly.py (hunk 3 above)
Comment (crankycoder) [reply dags/taar_weekly.py:9]: Thx.  Fixed in 49c1c930

File: dags/operators/moz_dataproc_operator.py (hunk 1 above)
Comment (crankycoder) [reply dags/operators/moz_dataproc_operator.py]: Added better doc strings: e04d7375

File: dags/utils/dataproc.py (hunk 2 above)
Comment (crankycoder) [reply dags/utils/dataproc.py:291]: Added better doc strings: e04d7375

File: dags/taar_weekly.py (hunk 4 above)
Comment (crankycoder) [reply dags/taar_weekly.py:13]: Fixed in 4363acb

Comment (crankycoder) [reply dags/taar_weekly.py:13]: Crap.  I had to revert in 4363acb7 as it broke the DAG definition in https://github.com/mozilla/telemetry-airflow/pull/929/files#diff-00957536695958afe22bc38f1597e7c4R71

**Pull request merged**